File: get-ime.py
#!/usr/bin/python
import objc
import Foundation

#
# Load carbon framework
#
carbon = {}

bundle = Foundation.NSBundle.bundleWithIdentifier_('com.apple.HIToolbox')
objc.loadBundleFunctions(bundle, carbon, [
    ('TISCopyCurrentKeyboardLayoutInputSource', '@@'),
    ('TISGetInputSourceProperty', '@@@'),
])
objc.loadBundleVariables(bundle, carbon, [
    ('kTISPropertyLocalizedName', '@'),
])

#
# Print IME Name
#
source = carbon['TISCopyCurrentKeyboardLayoutInputSource']()
if source != None:
    name = carbon['TISGetInputSourceProperty'](source, carbon['kTISPropertyLocalizedName'])
    print name

File: get-ime.swift
import Carbon

//
// Print IME Name
//
if let source = TISCopyCurrentKeyboardLayoutInputSource()?.takeUnretainedValue() {
  if let ref = TISGetInputSourceProperty(source, kTISPropertyLocalizedName) {
    let name = Unmanaged<CFString>.fromOpaque(ref).takeUnretainedValue() as String
    print(name)
  }
}
